### ISSUE TITLE
Release 4.6.0 - RC 1 - WPK upgrade tests 

### ISSUE THREAD

**Issue by cborla** [wazuh/wazuh#19527]
The following issue aims to run `upgrade WPK tests` for the current release candidate, report the results, and open new issues for any encountered errors.

## WPK upgrade test information
|                                          |                                            |
|------------------------------------------|--------------------------------------------|
| **Main release candidate issue**         | #19471 |
| **Version**                              | 4.6.0 |
| **Release candidate #**                  | RC 1 |
| **Tag**                                  | [v4.6.0-rc1](https://github.com/wazuh/wazuh/tree/v4.6.0-rc1) |
| **Previous WPK upgrade tests issue** | [#18875](https://github.com/wazuh/wazuh/issues/19213) |

## Test report procedure
All individual test checks must be marked as:
| |                                            |
|---------------------------------|--------------------------------------------|
| Pass | The test ran successfully. |
| Xfail | The test was expected to fail and it failed. It must be properly justified and reported in an issue.  |
| Skip | The test was not run. It must be properly justified and reported in an issue.  |
| Fail | The test failed. A new issue must be opened to evaluate and address the problem. |

All test results must have one the following statuses: 
| |                                            |
|---------------------------------|--------------------------------------------|
| 🟢 | All checks passed. |
| 🔴 | There is at least one failed check. |
| 🟡 | There is at least one expected fail or skipped test and no failures. |

Any failing test must be properly addressed with a new issue, detailing the error and the possible cause. It must be included in the `Fixes` section of the current release candidate main issue.

Any expected fail or skipped test must have an issue justifying the reason. All auditors must validate the justification for an expected fail or skipped test.

An extended report of the test results must be attached as a zip or txt. This report can be used by the auditors to dig deeper into any possible failures and details.

## Tests

To evaluate this feature, it is necessary to test upgrading the agent and also the case when the upgrade fails (rollback). The `tree` command will be used to compare, before and after the upgrade/rollback process, and check that the presence, ownership and permissions of the files and directories are expected.

Wazuh versions to test (Upgrade to the current agent version):
### Linux

|OS|From Version|Status|Upgrade fail|Upgrade OK| Comment |
|----|-----|------|---------------|------------------|------------------|
|[CentOS 8](https://github.com/wazuh/wazuh/issues/19527#issuecomment-1753487804)|3.13.6| 🟢| 🟡| 🟢| [Known Issue](https://github.com/wazuh/wazuh/issues/15587)|
|[CentOS 8](https://github.com/wazuh/wazuh/issues/19527#issuecomment-1753487804)|4.5.3| 🟢| 🟢| 🟢| |
|[CentOS 8](https://github.com/wazuh/wazuh/issues/19527#issuecomment-1753487804)|4.6.0| 🟢| 🟢| 🟢| |
|[Ubuntu 22](https://github.com/wazuh/wazuh/issues/19527#issuecomment-1753895353)|4.5.3| 🟢| 🟢| 🟢| |
|[Ubuntu 22](https://github.com/wazuh/wazuh/issues/19527#issuecomment-1753895353)|4.6.0| 🟢| 🟢| 🟢| |
|[openSUSE Tumbleweed](https://github.com/wazuh/wazuh/issues/19527#issuecomment-1756730982)|4.5.3| 🟢| 🟢| 🟢| |
|[openSUSE Tumbleweed](https://github.com/wazuh/wazuh/issues/19527#issuecomment-1756730982)|4.6.0| 🟢| 🟢| 🟢| |
|[Amazon Linux 2](https://github.com/wazuh/wazuh/issues/19527#issuecomment-1756143827)|4.5.3| 🟢| 🟢| 🟢| |
|[Amazon Linux 2](https://github.com/wazuh/wazuh/issues/19527#issuecomment-1756143827)|4.6.0| 🟢| 🟢| 🟢| |

### Windows

|OS|From Version|Status|Upgrade fail|Upgrade OK| Comment |
|----|-----|------|---------------|------------------|------------------|
|[Windows 11](https://github.com/wazuh/wazuh/issues/19527#issuecomment-1753722908)|4.5.3| 🟢 |🟢  | 🟢 | |
|[Windows 11](https://github.com/wazuh/wazuh/issues/19527#issuecomment-1753722908)|4.6.0| 🟢 | 🟢 | 🟢 | |
|[Server 2022](https://github.com/wazuh/wazuh/issues/19527#issuecomment-1756123210)|3.13.6| 🟢 | 🟡 | 🟢 |  [Known Issue](https://github.com/wazuh/wazuh/issues/15587)|
|[Server 2022](https://github.com/wazuh/wazuh/issues/19527#issuecomment-1756123210)|4.5.3| 🟢 | 🟢 | 🟢 | |
|[Server 2022](https://github.com/wazuh/wazuh/issues/19527#issuecomment-1756123210)|4.6.0| 🟢 | 🟢 | 🟢 | |

### macOS

|OS|From Version|Status|Upgrade fail|Upgrade OK| Comment |
|----|-----------|--------|-------|------------------|------------------|
| [Ventura](https://github.com/wazuh/wazuh/issues/19527#issuecomment-1753658184)|4.4.5| 🟢 |🟢| 🟢| |
| [Ventura](https://github.com/wazuh/wazuh/issues/19527#issuecomment-1753658184)|4.5.2| 🟢| 🟢| 🟢| |
| [Ventura](https://github.com/wazuh/wazuh/issues/19527#issuecomment-1753658184)|4.6.0| 🟢| 🟢| 🟢| |

## Conclusions

In conclusion, all tests were correct, no faults were found in the procedure, there is a different behavior for versions prior to 4.X.

| **Status** | **Test**    | **Failure type**    | **Notes**      |
|----------------|-------------|---------------------|----------------|
| 🟡  | From Wazuh-Agent 3.13.6 | The agent rollback is generated correctly, only the manager is waiting for a response that is implemented from 4.X onwards, is a known behavior. | [Known Issue](https://github.com/wazuh/wazuh/issues/15587) |

## Auditors validation
The definition of done for this one is the validation of the conclusions and the test results from all auditors.

All checks from below must be accepted in order to close this issue.

- [x] @cborla 
- [x] @JcabreraC 
- [x] @davidjiglesias

**Comment by cborla** [comment]
WPK fail package 

- [Job](https://ci.wazuh.info/view/Packages/job/Packages_builder_tier/3184/)
- Linux [wazuh_agent_v4.6.0-wpk.fail_linux_x86_64.wpk](https://packages-dev.wazuh.com/warehouse/test/4.6/wpk/linux/x86_64/wazuh_agent_v4.6.0-wpk.fail_linux_x86_64.wpk)
- Windows [wazuh_agent_v4.6.0-wpk.fail_windows.wpk](https://packages-dev.wazuh.com/warehouse/test/4.6/wpk/windows/wazuh_agent_v4.6.0-wpk.fail_windows.wpk)
- macOS [wazuh_agent_v4.6.0-wpk.fail_macos_x86_64.wpk](https://packages-dev.wazuh.com/warehouse/test/4.6/wpk/macos/x86_64/pkg/wazuh_agent_v4.6.0-wpk.fail_macos_x86_64.wpk)

**Comment by LucioDonda** [comment]
# TEST: CentOS 8 / RHEL 8

## From Wazuh-Agent 3.13.6 
###  UPGRADE FAIL: 🟡 

- [x] The wazuh home backup is restored correctly (no traces of the installation, but only the `.tar.gz` backup and the logs).
- [x] The permissions and owners of the following directories did NOT change:
  - `/`
  - `/var`
  - `/usr`, `/usr/lib/`, `/usr/lib/systemd/`, `/usr/lib/systemd/system/`
  - `/etc`, `/etc/systemd/`, `/etc/systemd/system/`, `/etc/rc.d`, `/etc/initd.d/`, `/etc/initd.d/rc.d/`
- [x] Wazuh service runs ossec-control (`systemctl cat wazuh-agent.service`)
- [x] The service was enabled (`systemctl is-enabled wazuh-agent.service`)
- [x] Init file runs wazuh-control (`cat /etc/rc.d/init.d/wazuh-agent`)
- [x] Wazuh as service is enabled `systemctl list-unit-files`
- [x] Wazuh starts and connects when the backup is restored (`cat /var/ossec/var/run/ossec-agentd.state`)
- [x] Wazuh starts and connects automatically when the system is rebooted.
- [x] Restore SELinux policies (`semodule -l | grep -i wazuh`)

The freeze time during upgrade, named in [this comment](https://github.com/wazuh/wazuh/issues/15529#issuecomment-1335136549) is still happening.

**Upgrade reports**
[3.13.6-Failed.zip](https://github.com/wazuh/wazuh/files/12849458/3.13.6-Failed.zip)

###  UPGRADE OK: 🟢 

- [x] Upgrade is performed successfully (agent conects to the manager after upgrading)
- [x] Service starts automatically after rebooting
- [x] Agent connects to the manager after rebooting

**Upgrade reports**
[3.13.16-Success.zip](https://github.com/wazuh/wazuh/files/12849459/3.13.16-Success.zip)

----------------------------------------------------------------------------------------------------------


## From Wazuh-Agent 4.5.3
###  UPGRADE FAIL: 

- [ ] The wazuh home backup is restored correctly (no traces of the installation, but only the `.tar.gz` backup and the logs).
- [ ] The permissions and owners of the following directories did NOT change:
  - `/`
  - `/var`
  - `/usr`, `/usr/lib/`, `/usr/lib/systemd/`, `/usr/lib/systemd/system/`
  - `/etc`, `/etc/systemd/`, `/etc/systemd/system/`, `/etc/rc.d`, `/etc/initd.d/`, `/etc/initd.d/rc.d/`
- [ ] Wazuh service runs wazuh-control (`systemctl cat wazuh-agent.service`)
- [ ] The service was enabled (`systemctl is-enabled wazuh-agent.service`)
- [ ] Init file runs wazuh-control (`cat /etc/rc.d/init.d/wazuh-agent`)
- [ ] Wazuh as service is enabled `chkconfig --list` (`chkconfig` unavailable)
- [ ] Wazuh starts and connects when the backup is restored (`cat /var/ossec/var/run/wazuh-agentd.state`)
- [ ] Wazuh starts and connects automatically when the system is rebooted.
- [ ] Restore SELinux policies (`semodule -l | grep -i wazuh`) (Disabled)

**Upgrade reports**


###  UPGRADE OK: 

- [ ] Upgrade is performed successfully (agent connects to the manager after upgrading)
- [ ] Service starts automatically after rebooting
- [ ] Agent connects to the manager after rebooting

**Upgrade reports**


----------------------------------------------------------------------------------------------------------


## From Wazuh-Agent 4.6.0
###  UPGRADE FAIL:

- [ ] The wazuh home backup is restored correctly (no traces of the installation, but only the `.tar.gz` backup and the logs).
- [ ] The permissions and owners of the following directories did NOT change:
  - `/`
  - `/var`
  - `/usr`, `/usr/lib/`, `/usr/lib/systemd/`, `/usr/lib/systemd/system/`
  - `/etc`, `/etc/systemd/`, `/etc/systemd/system/`, `/etc/rc.d`, `/etc/initd.d/`, `/etc/initd.d/rc.d/`
- [ ] Wazuh service runs wazuh-control (`systemctl cat wazuh-agent.service`)
- [ ] The service was enabled (`systemctl is-enabled wazuh-agent.service`)
- [ ] Init file runs wazuh-control (`cat /etc/rc.d/init.d/wazuh-agent`)
- [ ] Wazuh as service is enabled `chkconfig --list` (`chkconfig` unavailable)
- [ ] Wazuh starts and connects when the backup is restored (`cat /var/ossec/var/run/wazuh-agentd.state`)
- [ ] Wazuh starts and connects automatically when the system is rebooted.
- [ ] Restore SELinux policies (`semodule -l | grep -i wazuh`) (Disabled)

**Upgrade reports**

###  UPGRADE OK:

- [ ] Upgrade is performed successfully (agent connects to the manager after upgrading)
- [ ] Service starts automatically after rebooting
- [ ] Agent connects to the manager after rebooting

**Upgrade reports**

**Comment by nmkoremblum** [comment]
# TEST: macOS Ventura (13.5.2)

## From Wazuh-Agent 4.4.5
###  UPGRADE FAIL: :yellow_circle: 

- [x] The wazuh home backup is restored correctly (no traces of the installation, but only the `.tar.gz` backup and the logs).
- [x] The permissions and owners of the following directories did NOT change:
  - `/Library/LaunchDaemons/` `/Library/StartupItems/` `/Library/StartupItems/WAZUH/` `/Library/LaunchDaemons/` 
- [x] Wazuh service is enabled and active (`launchctl print system/com.wazuh.agent`)
- [x] Wazuh starts and connects when the backup is restored (`cat /var/ossec/var/run/wazuh-agentd.state`)
- [x] Wazuh starts and connects automatically when the system is rebooted.

###  UPGRADE OK: :yellow_circle: 

- [x] Upgrade is performed successfully (agent connects to the manager after upgrading)
- [x] Service starts automatically after rebooting
- [x] Agent connects to the manager after rebooting

**Upgrade reports**


---

## From Wazuh-Agent 4.5.2
###  UPGRADE FAIL: :yellow_circle: 

- [x] The wazuh home backup is restored correctly (no traces of the installation, but only the `.tar.gz` backup and the logs).
- [x] The permissions and owners of the following directories did NOT change:
  - `/Library/LaunchDaemons/` `/Library/StartupItems/` `/Library/StartupItems/WAZUH/` `/Library/LaunchDaemons/` 
- [x] Wazuh service is enabled and active (`launchctl print system/com.wazuh.agent`)
- [x] Wazuh starts and connects when the backup is restored (`cat /var/ossec/var/run/wazuh-agentd.state`)
- [x] Wazuh starts and connects automatically when the system is rebooted.

###  UPGRADE OK: :yellow_circle: 

- [x] Upgrade is performed successfully (agent connects to the manager after upgrading)
- [x] Service starts automatically after rebooting
- [x] Agent connects to the manager after rebooting

**Upgrade reports**


---

## From Wazuh-Agent 4.6.0
###  UPGRADE FAIL: :yellow_circle: 

- [x] The wazuh home backup is restored correctly (no traces of the installation, but only the `.tar.gz` backup and the logs).
- [x] The permissions and owners of the following directories did NOT change:
  - `/Library/LaunchDaemons/` `/Library/StartupItems/` `/Library/StartupItems/WAZUH/` `/Library/LaunchDaemons/` 
- [x] Wazuh service is enabled and active (`launchctl print system/com.wazuh.agent`)
- [x] Wazuh starts and connects when the backup is restored (`cat /var/ossec/var/run/wazuh-agentd.state`)
- [x] Wazuh starts and connects automatically when the system is rebooted.

###  UPGRADE OK: :yellow_circle: 

- [x] Upgrade is performed successfully (agent connects to the manager after upgrading)
- [x] Service starts automatically after rebooting
- [x] Agent connects to the manager after rebooting

**Upgrade reports**

**Comment by nbertoldo** [comment]
# TEST: Windows 11

## From Wazuh-Agent 4.5.3
###  UPGRADE FAIL: :green_circle:

- [x]  Wazuh-Agent folder tree: No files are lost after the rollback. The logs of the failed upgrade (ossec.log) are kept.
- [x]  After the rollback the agent connects to the manager
- [x]  After reboot, the Wazuh-Agent starts and connects to the manager.
- [x]  The correct Wazuh-Agent version is shown in the list of Windows' programs and features.
- [x]  A new version of Wazuh-Agent can be manually installed via MSI after the rollback process.

###  UPGRADE OK: :green_circle:

- [x] Message Upgrade finished successfully. is shown in upgrade.log file.
- [x]  Wazuh service is started and the agent is connected to the manager.
- [x]  The version shown in the control panel is 4.6.0
- [x]  The new SCA policy is added and executed

**Upgrade report**
[report_Windows_11_v4.5.3.zip](https://github.com/wazuh/wazuh/files/12848699/report_Windows_11_v4.5.3.zip)


## From Wazuh-Agent 4.6.0
###  UPGRADE FAIL: :green_circle:

- [x]  Wazuh-Agent folder tree: No files are lost after the rollback. The logs of the failed upgrade (ossec.log) are kept.
- [x]  After the rollback the agent connects to the manager
- [x]  After reboot, the Wazuh-Agent starts and connects to the manager.
- [x]  The correct Wazuh-Agent version is shown in the list of Windows' programs and features.
- [x]  A new version of Wazuh-Agent can be manually installed via MSI after the rollback process.

###  UPGRADE OK: :green_circle:

- [x] Message Upgrade finished successfully. is shown in upgrade.log file.
- [x]  Wazuh service is started and the agent is connected to the manager.
- [x]  The version shown in the control panel is 4.6.0
- [x]  The new SCA policy is added and executed

**Upgrade report**
[report_Windows_11_v4.6.0.zip](https://github.com/wazuh/wazuh/files/12850321/report_Windows_11_v4.6.0.zip)

**Comment by LucioDonda** [comment]
# TEST: Ubuntu 22

----------------------------------------------------------------------------------------------------------
## From Wazuh-Agent v4.5.3
###  UPGRADE FAIL  🟢

- [x] The wazuh home backup is restored correctly (no traces of the installation, but only the `.tar.gz` backup and the logs).
- [x] The permissions and owners of the following directories did NOT change:
  - `/`
  - `/var`
  - `/usr`, `/usr/lib/`, `/usr/lib/systemd/`, `/usr/lib/systemd/system/`
  - `/etc`, `/etc/systemd/`, `/etc/systemd/system/`, `/etc/rc.d`, `/etc/initd.d/`, `/etc/initd.d/rc.d/`
- [x] Wazuh service runs wazuh-control (`systemctl cat wazuh-agent.service`)
- [x] The service was enabled (`systemctl is-enabled wazuh-agent.service`)
- [x] Init file runs wazuh-control (`cat /etc/init.d/wazuh-agent`)
- [x] Wazuh starts and connects when the backup is restored (`cat /var/ossec/var/run/wazuh-agentd.state`)
- [x] Wazuh starts and connects automatically when the system is rebooted.

**Upgrade reports**
[Failed-4.5.3.zip](https://github.com/wazuh/wazuh/files/12850700/Failed-4.5.3.zip)

###  UPGRADE OK  🟢

- [x] Upgrade is performed successfully (agent connects to the manager after upgrading)
- [x] Service starts automatically after rebooting
- [x] Agent connects to the manager after rebooting

**Upgrade reports**
[Success-4.5.3.zip](https://github.com/wazuh/wazuh/files/12850701/Success-4.5.3.zip)

----------------------------------------------------------------------------------------------------------

## From Wazuh-Agent v4.6.0
###  UPGRADE FAIL  

- [ ] The wazuh home backup is restored correctly (no traces of the installation, but only the `.tar.gz` backup and the logs).
- [ ] The permissions and owners of the following directories did NOT change:
  - `/`
  - `/var`
  - `/usr`, `/usr/lib/`, `/usr/lib/systemd/`, `/usr/lib/systemd/system/`
  - `/etc`, `/etc/systemd/`, `/etc/systemd/system/`, `/etc/rc.d`, `/etc/initd.d/`, `/etc/initd.d/rc.d/`
- [ ] Wazuh service runs wazuh-control (`systemctl cat wazuh-agent.service`)
- [ ] The service was enabled (`systemctl is-enabled wazuh-agent.service`)
- [ ] Init file runs wazuh-control (`cat /etc/systemd/system/wazuh-agent.service`)
- [ ] Wazuh starts and connects when the backup is restored (`cat /var/ossec/var/run/wazuh-agentd.state`)
- [ ] Wazuh starts and connects automatically when the system is rebooted.

**Upgrade reports**

###  UPGRADE OK  

- [ ] Upgrade is performed successfully (agent connects to the manager after upgrading)
- [ ] Service starts automatically after rebooting
- [ ] Agent connects to the manager after rebooting

**Upgrade reports**

**Comment by nbertoldo** [comment]
# TEST: Windows Server 2022

## From Wazuh-Agent 3.13.6
###  UPGRADE FAIL: :yellow_circle:

- [x] Wazuh-Agent folder tree: No files are lost after the rollback. The logs of the failed upgrade (ossec.log) are kept.
- [x] After the rollback the agent connects to the manager
- [x] After reboot, the Wazuh-Agent starts and connects to the manager.
- [x] The correct Wazuh-Agent version is shown in the list of Windows' programs and features.
- [x] A new version of Wazuh-Agent can be manually installed via MSI after the rollback process.

**Note** Found same unexpected behavior as:
https://github.com/wazuh/wazuh/issues/15529#issuecomment-1335136549

###  UPGRADE OK: :green_circle:

- [x] Message Upgrade finished successfully. is shown in upgrade.log file.
- [x] Wazuh service is started and the agent is connected to the manager.
- [x] The version shown in the control panel is 4.6.0
- [x] The new SCA policy is added and executed

**Upgrade report**
[report_Windows_Server_2022_v3.13.6.zip](https://github.com/wazuh/wazuh/files/12859488/report_Windows_Server_2022_v3.13.6.zip)

## From Wazuh-Agent 4.5.3
###  UPGRADE FAIL: :green_circle:

- [x]  Wazuh-Agent folder tree: No files are lost after the rollback. The logs of the failed upgrade (ossec.log) are kept.
- [x]  After the rollback the agent connects to the manager
- [x]  After reboot, the Wazuh-Agent starts and connects to the manager.
- [x]  The correct Wazuh-Agent version is shown in the list of Windows' programs and features.
- [x]  A new version of Wazuh-Agent can be manually installed via MSI after the rollback process.

###  UPGRADE OK: :green_circle:

- [x] Message Upgrade finished successfully. is shown in upgrade.log file.
- [x]  Wazuh service is started and the agent is connected to the manager.
- [x]  The version shown in the control panel is 4.6.0
- [x]  The new SCA policy is added and executed

**Upgrade report**
[report_Windows_Server_2022_v4.5.3.zip](https://github.com/wazuh/wazuh/files/12860141/report_Windows_Server_2022_v4.5.3.zip)


## From Wazuh-Agent 4.6.0
###  UPGRADE FAIL: :green_circle:

- [x]  Wazuh-Agent folder tree: No files are lost after the rollback. The logs of the failed upgrade (ossec.log) are kept.
- [x]  After the rollback the agent connects to the manager
- [x]  After reboot, the Wazuh-Agent starts and connects to the manager.
- [x]  The correct Wazuh-Agent version is shown in the list of Windows' programs and features.
- [x]  A new version of Wazuh-Agent can be manually installed via MSI after the rollback process.

###  UPGRADE OK: :green_circle:

- [x] Message Upgrade finished successfully. is shown in upgrade.log file.
- [x]  Wazuh service is started and the agent is connected to the manager.
- [x]  The version shown in the control panel is 4.6.0
- [x]  The new SCA policy is added and executed

**Upgrade report**
[report_Windows_Server_2022_v4.6.0.zip](https://github.com/wazuh/wazuh/files/12861316/report_Windows_Server_2022_v4.6.0.zip)

**Comment by JorgeMarinoDev** [comment]
# TEST: Amazon Linux 2 

Vagrant Machine : [vagrantcloud](https://app.vagrantup.com/gbailey/boxes/amzn2)
**Amazon Linux 2 (2.0.20230926.0) with VirtualBox 7.0.10 Guest Additions and Amazon Linux 2 updates as of October 4, 2023.**

## From Wazuh-Agent 4.5.3
###  UPGRADE FAIL: 🟢

- [x] The wazuh home backup is restored correctly (no traces of the installation, but only the `.tar.gz` backup and the logs).
- [x] The permissions and owners of the following directories did NOT change:
  - `/`
  - `/var`
  - `/usr`, `/usr/lib/`, `/usr/lib/systemd/`, `/usr/lib/systemd/system/`
  - `/etc`, `/etc/systemd/`, `/etc/systemd/system/`, `/etc/rc.d`, `/etc/initd.d/`, `/etc/initd.d/rc.d/`
- [x] Wazuh service runs wazuh-control (`systemctl cat wazuh-agent.service`)
- [x] The service was enabled (`systemctl is-enabled wazuh-agent.service`)
- [x] Init file runs wazuh-control (`cat /etc/rc.d/init.d/wazuh-agent`)
- [x] Wazuh as service is enabled `systemctl list-unit-files`
- [x] Wazuh starts and connects when the backup is restored (`cat /var/ossec/var/run/wazuh-agentd.state`)
- [x] Wazuh starts and connects automatically when the system is rebooted.
- [x] Restore SELinux policies (`semodule -l | grep -i wazuh`) (Disabled)

###  UPGRADE OK: 🟢

- [x] Upgrade is performed successfully (agent connects to the manager after upgrading)
- [x] Service starts automatically after rebooting
- [x] Agent connects to the manager after rebooting

**Upgrade reports**
[453_460.zip](https://github.com/wazuh/wazuh/files/12861418/453_460.zip)

----------------------------------------------------------------------------------------------------------


## From Wazuh-Agent 4.6.0
###  UPGRADE FAIL: 🟢

- [x] The wazuh home backup is restored correctly (no traces of the installation, but only the `.tar.gz` backup and the logs).
- [x] The permissions and owners of the following directories did NOT change:
  - `/`
  - `/var`
  - `/usr`, `/usr/lib/`, `/usr/lib/systemd/`, `/usr/lib/systemd/system/`
  - `/etc`, `/etc/systemd/`, `/etc/systemd/system/`, `/etc/rc.d`, `/etc/initd.d/`, `/etc/initd.d/rc.d/`
- [x] Wazuh service runs wazuh-control (`systemctl cat wazuh-agent.service`)
- [x] The service was enabled (`systemctl is-enabled wazuh-agent.service`)
- [x] Init file runs wazuh-control (`cat /etc/rc.d/init.d/wazuh-agent`)
- [x] Wazuh as service is enabled `systemctl list-unit-files`
- [x] Wazuh starts and connects when the backup is restored (`cat /var/ossec/var/run/wazuh-agentd.state`)
- [x] Wazuh starts and connects automatically when the system is rebooted.
- [x] Restore SELinux policies (`semodule -l | grep -i wazuh`)

###  UPGRADE OK: 🟢

- [x] Upgrade is performed successfully (agent connects to the manager after upgrading)
- [x] Service starts automatically after rebooting
- [x] Agent connects to the manager after rebooting

**Upgrade reports**
[460_460.zip](https://github.com/wazuh/wazuh/files/12861420/460_460.zip)

**Comment by nmkoremblum** [comment]
# TEST: openSUSE Tumbleweed

## From Wazuh-Agent v4.5.3
###  UPGRADE FAIL :green_circle: 

- [X] The wazuh home backup is restored correctly (no traces of the installation, but only the `.tar.gz` backup and the logs).
- [X] The permissions and owners of the following directories did NOT change:
  - `/`
  - `/var`
  - `/usr`, `/usr/lib/`, `/usr/lib/systemd/`, `/usr/lib/systemd/system/`
  - `/etc`, `/etc/systemd/`, `/etc/systemd/system/`, `/etc/rc.d`, `/etc/initd.d/`, `/etc/initd.d/rc.d/`
- [X] Wazuh service runs wazuh-control (`systemctl cat wazuh-agent.service`)
- [ ] ~Wazuh service runs ossec-control (`systemctl cat wazuh-agent.service`)~
- [X] The service was enabled (`systemctl is-enabled wazuh-agent.service`)
- [X] Init file runs wazuh-control (`cat /etc/rc.d/init.d/wazuh-agent`)
- [ ] ~Init file runs ossec-control (`cat /etc/rc.d/init.d/wazuh-agent`)~
- [ ] ~Wazuh as service is enabled `chkconfig --list`~ (`chkconfig` unavailable)
- [X] Wazuh starts and connects when the backup is restored (`cat /var/ossec/var/run/wazuh-agentd.state`)
- [ ] ~Wazuh starts and connects when the backup is restored (`cat /var/ossec/var/run/ossec-agentd.state`)~
- [X] Wazuh starts and connects automatically when the system is rebooted.
- [ ] ~Restore SELinux policies (`semodule -l | grep -i wazuh`)~ (Disabled)

###  UPGRADE OK  :green_circle: 

- [X] Upgrade is performed successfully (agent connects to the manager after upgrading)
- [X] Service starts automatically after rebooting
- [X] Agent connects to the manager after rebooting

**Upgrade reports**

---
## From Wazuh-Agent v4.6.0
###  UPGRADE FAIL  :green_circle: 

- [X] The wazuh home backup is restored correctly (no traces of the installation, but only the `.tar.gz` backup and the logs).
- [X] The permissions and owners of the following directories did NOT change:
  - `/`
  - `/var`
  - `/usr`, `/usr/lib/`, `/usr/lib/systemd/`, `/usr/lib/systemd/system/`
  - `/etc`, `/etc/systemd/`, `/etc/systemd/system/`, `/etc/rc.d`, `/etc/initd.d/`, `/etc/initd.d/rc.d/`
- [X] Wazuh service runs wazuh-control (`systemctl cat wazuh-agent.service`)
- [ ] ~Wazuh service runs ossec-control (`systemctl cat wazuh-agent.service`)~
- [X] The service was enabled (`systemctl is-enabled wazuh-agent.service`)
- [X] Init file runs wazuh-control (`cat /etc/rc.d/init.d/wazuh-agent`)
- [ ] ~Init file runs ossec-control (`cat /etc/rc.d/init.d/wazuh-agent`)~
- [ ] ~Wazuh as service is enabled `chkconfig --list`~ (`chkconfig` unavailable)
- [X] Wazuh starts and connects when the backup is restored (`cat /var/ossec/var/run/wazuh-agentd.state`)
- [ ] ~Wazuh starts and connects when the backup is restored (`cat /var/ossec/var/run/ossec-agentd.state`)~
- [X] Wazuh starts and connects automatically when the system is rebooted.
- [ ] ~Restore SELinux policies (`semodule -l | grep -i wazuh`)~ (Disabled)

###  UPGRADE OK  :green_circle: 

- [X] Upgrade is performed successfully (agent connects to the manager after upgrading)
- [X] Service starts automatically after rebooting
- [X] Agent connects to the manager after rebooting

**Upgrade reports**
